src/test/app/EscrowToken_test.cpp

```diff
@@ -0,0 +1,2708 @@
+            env.close();
+            env.trust(USD(100'000), alice);
+            env.trust(USD(100'000), bob);
+            env.close();
+            env(pay(gw, alice, USD(10'000)));
+            env(pay(gw, bob, USD(10'000)));
+            env.close();
+
+            // alice can create escrow w/ xfer rate
+            auto const preAlice = env.balance(alice, USD);
+            auto const seq1 = env.seq(alice);
+            auto const delta = USD(125);
+            env(escrow::create(alice, bob, delta),
+                escrow::condition(escrow::cb1),
+                escrow::finish_time(env.now() + 1s),
+                fee(baseFee * 150));
+            env.close();
+            auto transferRate = escrow::rate(env, alice, seq1);
+            BEAST_EXPECT(
+                transferRate.value == std::uint32_t(1'000'000'000 * 1.25));
+
+            // issuer changes rate lower
+            env(rate(gw, 1.00));
+            env.close();
+
+            // bob can finish escrow - rate changed
+            env(escrow::finish(bob, alice, seq1),
+                escrow::condition(escrow::cb1),
+                escrow::fulfillment(escrow::fb1),
+                fee(baseFee * 150));
+            env.close();
+
+            BEAST_EXPECT(env.balance(alice, USD) == preAlice - delta);
+            BEAST_EXPECT(env.balance(bob, USD) == USD(10125));
+        }
+    }
+
+    void
+    testIOULimitAmount(FeatureBitset features)
+    {
+        testcase("IOU Limit");
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const gw = Account{"gateway"};
+        auto const USD = gw["USD"];
+
+        // test LimitAmount
+        {
+            Env env{*this, features};
+            auto const baseFee = env.current()->fees().base;
+            env.fund(XRP(1'000), alice, bob, gw);
+            env(fset(gw, asfAllowTokenLocking));
+            env.close();
+            env.trust(USD(10'000), alice, bob);
+            env.close();
+            env(pay(gw, alice, USD(1'000)));
+            env(pay(gw, bob, USD(1'000)));
+            env.close();
+
+            // alice can create escrow
+            auto seq1 = env.seq(alice);
+            auto const delta = USD(125);
+            env(escrow::create(alice, bob, delta),
+                escrow::condition(escrow::cb1),
+                escrow::finish_time(env.now() + 1s),
+                fee(baseFee * 150));
+            env.close();
+
+            // bob can finish
+            auto const preBobLimit = env.limit(bob, USD);
+            env(escrow::finish(bob, alice, seq1),
+                escrow::condition(escrow::cb1),
+                escrow::fulfillment(escrow::fb1),
+                fee(baseFee * 150));
+            env.close();
+            auto const postBobLimit = env.limit(bob, USD);
+            // bobs limit is NOT changed
+            BEAST_EXPECT(postBobLimit == preBobLimit);
```

Comment:
what would happen if bob lowers his limit to be lower than the amount that he's going to receive? can we add a test for that?

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5185/commits/39854ae791586a5a45a08e9b8ae9795ac94c78d9